task1/task1.js

```diff
@@ -0,0 +1,15 @@
+const reverseString = (str) => {
+    const words = str.split(/\W|_/).reverse();
+
+    const separatorsResult = str.matchAll(/\W|_/g);
+    const separators = Array.from(separatorsResult);
+
+
+    let result = '';
+    for(let i = 0; i < words.length; i++) {
+        result = result + words[i].split('').reverse().join('') + (i < separators.length ?  separators[i] : '');
+    }
+    return result
+}
+
+console.log(reverseString('a,b$c_d'));
```

Comment:
@elenagronskaya your implementation should work both for `Ab,c,de$!` and `a,b$c` exactly